api/src/search/backend/load_opportunities_to_index.py

```diff
@@ -347,17 +344,15 @@ def get_attachment_json_for_opportunity(
         attachments = []
         for att in opp_attachments:
             if self.filter_attachment(att):
-                with file_util.open_stream(
-                    att.file_location,
-                    "rb",
-                ) as file:
-                    file_content = file.read()
-                    attachments.append(
-                        {
-                            "filename": att.file_name,
-                            "data": base64.b64encode(file_content).decode("utf-8"),
-                        }
-                    )
+                file_text = extract_text_from_file(att.file_location)
+                attachments.append(
+                    {
+                        "filename": att.file_name,
+                        "attachment": {
+                            "content": file_text,
+                        },
+                    }
+                )
```

Comment:
Two thoughts here:
1. We should have a try/except wrapping this - if a single attachment is bad (I'm going to guess at least some are misnamed), we shouldn't break the process, just log a warning/increment a counter and continue.
2. We might want to consider adding a limit to the size of the response from the extract logic. A 200 page PDF we might be fine cutting off at a certain point - the ingest attachment stuff we're replacing had a limit of 100,000 by default: https://docs.opensearch.org/docs/latest/install-and-configure/additional-plugins/ingest-attachment-plugin/